apps/foundation-demo/src/app/checkout/components/shipping-form/shipping-form.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { FormBuilder, FormGroup, Validators, AbstractControl } from '@angular/forms';
+
+import { ShippingAddress } from '@daffodil/core';
+
+@Component({
+  selector: 'shipping-form',
+  templateUrl: './shipping-form.component.html',
+  styleUrls: ['./shipping-form.component.scss']
+})
+export class ShippingFormComponent implements OnInit {
+
+  @Input() shippingInfo: ShippingAddress;
+  @Output() updateShippingInfo: EventEmitter<any> = new EventEmitter();
+
+  form: FormGroup
+  firstname: AbstractControl;
+  lastname: AbstractControl;
+  street: AbstractControl;
+  city: AbstractControl;
+  state: AbstractControl;
+  postcode: AbstractControl;
+  telephone: AbstractControl;
+
+  constructor(
+    private fb: FormBuilder
+  ) { }
+
+  ngOnInit() {
+    this.form = this.fb.group({
+      'firstname': [this.shippingInfo ? this.shippingInfo.firstname : '', Validators.required],
+      'lastname': [this.shippingInfo ? this.shippingInfo.lastname : '', Validators.required],
+      'street': [this.shippingInfo ? this.shippingInfo.street : '', Validators.required],
+      'city': [this.shippingInfo ? this.shippingInfo.city : '', Validators.required],
+      'state': [this.shippingInfo ? this.shippingInfo.state : 'State', Validators.required],
+      'postcode': [this.shippingInfo ? this.shippingInfo.postcode : '', Validators.required],
+      'telephone': [this.shippingInfo ? this.shippingInfo.telephone : '', Validators.required]
+    });
+
+    this.firstname = this.form.controls['firstname'];
```

Comment:
Are these child variables necessary?

Reply:
Their purpose is just to make the html more terse. Without these, the html goes from

```
<input type="text" class="shipping-form__first-name"
    placeholder="First Name"
    name="firstname"
    [formControl]="firstname"
    [class.shipping-form__error]="firstname.errors && (firstname.touched || ngForm.submitted)"
    [class.shipping-form__valid]="!firstname.errors && firstname.touched"/>
```
to 
```
<input type="text" class="shipping-form__first-name"
    placeholder="First Name"
    name="firstname"
    [formControl]="form.controls.firstname"
    [class.shipping-form__error]="form.controls.firstname.errors && (form.controls.firstname.touched || ngForm.submitted)"
    [class.shipping-form__valid]="!form.controls.firstname.errors && form.controls.firstname.touched"/>
```

Though it's possible most of those references will be cut out when I refactor input into our own custom module

Reply:
with the input refactor, there's only one reference to the formControl now, so I removed all of the child variables